### PR TITLE
feat: generate and package licenses.html file

### LINKED ISSUE #36: Generate license file from dependencies
We should generate this file based on our dependencies — perhaps using [`cargo-about`](https://crates.io/crates/cargo-about)?) — and include this file in the packaged distributions.  Our `LICENSE` should also be included.

### PR DESCRIPTION
**Pull request by o0Ignition0o**
feat: generate and package licenses.html file

fixes #36 

This commit allows us to generate a licenses.html, file, that contains all of the licenses our dependencies use. It is based on cargo-about:
- update `xtask check-compliance` to run cargo-about alongside cargo-deny, and bail if the licenses.html fils is not up to date.
- update CI to install `cargo-about` when checking for lints / compliance.
- update DEVELOPMENT.md to let users know how to run `cargo xtask lint` and `cargo xtask check-compliance`, as well as how to install the required dependencies.
- add `licenses.html` to the tarballs when building a release artefact.

### CODE REVIEW

**Comment by o0Ignition0o** [comment]
Here's the output if the licenses.html file is not in sync:

image

File: xtask/src/commands/check.rs [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
+            "about.hbs"
+        ]);
+
+        let licences_html_after = read_to_string(LICENSES_HTML_PATH)?;
+
+        ensure!(
+            licenses_html_before == licences_html_after,
+            r#"🚨🚨🚨 licenses.html file is not up to date. 🚨🚨🚨
+please run `cargo about generate --workspace -o licenses.html about.hbs` to generate an up to date licenses list, and check the file in to the repository. 
+        "#
```
Comment (cecton) [inline xtask/src/commands/check.rs:36]: You don't seem to know the trick with the `\` at the end of the line

```suggestion
            r#"🚨 licenses.html file is not up to date. 🚨\n\
            please run `cargo about generate --workspace -o licenses.html about.hbs` to generate an up to date licenses list, and check the file in to the repository."#
```

File: xtask/src/commands/check.rs [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+static LICENSES_HTML_PATH: &str = "licenses.html";
+
 #[derive(Debug, StructOpt)]
 pub struct Compliance {}
 
 impl Compliance {
     pub fn run(&self) -> Result<()> {
         cargo!(["deny", "-L", "error", "check"]);
 
+        eprintln!("checking generated licenses.html file");
```
Comment (cecton) [inline xtask/src/commands/check.rs:15]:
```suggestion
        eprintln!("Checking generated licenses.html file...");
```

File: xtask/src/commands/check.rs [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
+            "-L",
+            "error",
+            "generate",
+            "--workspace",
+            "-o",
+            "licenses.html",
+            "about.hbs"
+        ]);
+
+        let licences_html_after = read_to_string(LICENSES_HTML_PATH)?;
```
Comment (cecton) [inline xtask/src/commands/check.rs:30]: If the file gets too big you might want to hash it instead. But I'm ok with this for now.

File: xtask/src/commands/check.rs [hunk 4]
```diff
@@ -1,14 +1,40 @@
...
+
+        cargo!([
+            "about",
+            "-L",
+            "error",
+            "generate",
+            "--workspace",
+            "-o",
+            "licenses.html",
+            "about.hbs"
```
Comment (cecton) [inline xtask/src/commands/check.rs:27]:
```suggestion
            "about.hbs",
```

**Review by cecton — APPROVED** [review]

File: xtask/src/commands/check.rs (hunk 1 above)
Comment (o0Ignition0o) [reply xtask/src/commands/check.rs:36]: Indeed! :o this is super cool :D

File: xtask/src/commands/check.rs (hunk 3 above)
Comment (o0Ignition0o) [reply xtask/src/commands/check.rs:30]: true, especially since I don't show any kind of diff. let's keep this in mind!

Comment (o0Ignition0o) [reply xtask/src/commands/check.rs]: Ok it's up to date now!

File: xtask/src/commands/check.rs [hunk 5]
```diff
@@ -1,14 +1,51 @@
...
     }
+
+    fn digest_for_license_file() -> Digest {
+        let mut ctx = ring::digest::Context::new(&ring::digest::SHA256);
+
+        ctx.update(
+            std::fs::read(LICENSES_HTML_PATH)
+                .expect("couldn't read file contents")
+                .as_slice(),
+        );
```
Comment (cecton) [inline xtask/src/commands/check.rs:47]: erhh that's a big fail :rofl:

**Pull request merged**